ccl/crosscluster/physical: TestStreamingRegionalConstraint failed

ccl/crosscluster/physical.TestStreamingRegionalConstraint [failed](https://tanzanite.cluster.engflow.com/invocations/default/57247f18-007b-4914-a7e6-16ba8f5db3d0?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9waHlzaWNhbDpwaHlzaWNhbF90ZXN0) on release-24.2.1-rc @ [23a8ffc28d2f31eef0cd13eca69693dc249d1196](https://github.com/cockroachdb/cockroach/commits/23a8ffc28d2f31eef0cd13eca69693dc249d1196):


```
=== RUN   TestStreamingRegionalConstraint
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStreamingRegionalConstraint1898422763
    test_log_scope.go:81: use -show-logs to present logs inline
    replication_stream_e2e_test.go:1386: -- test log scope end --
test logs left over in: outputs.zip/logTestStreamingRegionalConstraint1898422763
--- FAIL: TestStreamingRegionalConstraint (182.16s)
    --- FAIL: TestStreamingRegionalConstraint/fromSys=true (102.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamingRegionalConstraint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41615

--- a/pkg/ccl/crosscluster/physical/replication_stream_e2e_test.go
+++ b/pkg/ccl/crosscluster/physical/replication_stream_e2e_test.go
@@ -1310,6 +1310,7 @@ func TestStreamingRegionalConstraint(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t, "takes too long under race")
+	skip.UnderDeadlock(t, "takes too long under deadlock")
 
 	testutils.RunTrueAndFalse(t, "fromSys", func(t *testing.T, sys bool) {
 		ctx := context.Background()

